### PR TITLE
Make per-microservice/agent log levels work again

### PR DESCRIPTION
**Pull request by eriknordmark**
This stopped working when we introduced a single zedbox process since the logrus.SetLevel affects the global logrus.StandardLogger. 
So we introduce a logrus.Logger which we create per service, while still keeping the per agent base.LogObject which has the agentName and agentPid fields.

As part of this we make zedbox call agentlog.Init() for all cases, and pass that into the agent's Run() function.

The two first commits are cleanup in preparation and can be reviewed separately.

### CODE REVIEW

File: pkg/pillar/agentlog/agentlog.go [hunk 1]
```diff
@@ -547,29 +547,34 @@ func spoofStdFDs(log *base.LogObject, agentName string) *os.File {
...
+	hook2.agentPid = agentPid
+	logger.AddHook(hook2)
+	hook3 := new(SkipCallerHook)
+	logger.AddHook(hook3)
+	// For every separate process we set up output redirection
+	// to /persist/agentdebug (while keeping logs on stdout) and
+	// signal handlers
 	if once() {
-		logrus.SetOutput(os.Stdout)
+		logger.SetOutput(os.Stdout)
```
Comment (gkodali-zededa) [inline pkg/pillar/agentlog/agentlog.go:575]: Why is this line required?

Comment (eriknordmark) [reply pkg/pillar/agentlog/agentlog.go:575]: I don't know; I just renamed it.

Comment (eriknordmark) [reply pkg/pillar/agentlog/agentlog.go]: I removed it
